toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -541,21 +543,37 @@ func CreateSinglePartition(diskDevPath string, partitionNumber int, partitionTab
 	logger.Log.Debugf("Input partition start: %d, aligned start sector: %d", partition.Start, start)
 	logger.Log.Debugf("Input partition end: %d, end sector: %d", partition.End, end)
 
-	fsType := partition.FsType
+	mkpartArgs := []string{"--timeout", timeoutInSeconds, diskDevPath, "parted", diskDevPath, "--script", "mkpart"}
 
-	if end == 0 {
-		_, stderr, err := shell.Execute("flock", "--timeout", timeoutInSeconds, diskDevPath, "parted", diskDevPath, "--script", "mkpart", partType, fsType, fmt.Sprintf(sFmt, start), fillToEndOption)
-		if err != nil {
-			err = fmt.Errorf("failed to create partition using parted:\n%v\n%w", stderr, err)
-			return "", err
+	switch partitionTableType {
+	case configuration.PartitionTableTypeMbr:
+		// Part type.
+		mkpartArgs = append(mkpartArgs, partType)
+
+	case configuration.PartitionTableTypeGpt:
+		// Partition label.
+		if partition.Name == "" {
+			// Since parted is a scripting language, you have to specify "" to represent an empty string.
+			mkpartArgs = append(mkpartArgs, "\"\"")
+		} else {
+			mkpartArgs = append(mkpartArgs, partition.Name)
```

Comment:
Not 100% sure this works, but reads a bit better to me.
Also replaced "\\"\\"" -> \`""\`
```suggestion
			mkpartArgs = append(mkpartArgs, []string{"--name", `""`}...)
		} else {
			mkpartArgs = append(mkpartArgs, []string{"--name",partition.Name}...)
```

Reply:
Fixed.